api/v1alpha1/webhook_helpers.go

```diff
@@ -0,0 +1,281 @@
+		))
+	}
+
+	if _, isPresent := api.ClosedSourceDatabaseTypes[instance.Type]; isPresent {
+		if instance.Profiles == &(Profiles{}) || instance.Profiles.Software == (Profile{}) {
+			*errors = append(*errors, field.Invalid(instancePath.Child("profiles").Child("software"), instance.Profiles.Software, "Software Profile must be provided for the closed-source database engines"))
+		}
+	}
+
+	// validating time machine info
+	dailySnapshotTimeRegex := regexp.MustCompile(`^(2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9]$`)
+	if isMatch := dailySnapshotTimeRegex.MatchString(tmInfo.DailySnapshotTime); !isMatch {
+		*errors = append(*errors, field.Invalid(tmPath.Child("dailySnapshotTime"), tmInfo.DailySnapshotTime, "Invalid time format for the daily snapshot time. Use the 24-hour format (HH:MM:SS)."))
+	}
+
+	if tmInfo.SnapshotsPerDay < 1 || tmInfo.SnapshotsPerDay > 6 {
+		*errors = append(*errors, field.Invalid(tmPath.Child("snapshotsPerDay"), tmInfo.SnapshotsPerDay, "Number of snapshots per day should be within 1 to 6"))
+	}
+
+	if _, isPresent := api.AllowedLogCatchupFrequencyInMinutes[tmInfo.LogCatchUpFrequency]; !isPresent {
+		*errors = append(*errors, field.Invalid(tmPath.Child("logCatchUpFrequency"), tmInfo.LogCatchUpFrequency,
+			fmt.Sprintf("Log catchup frequency must be specified. Valid values are: %s", reflect.ValueOf(api.AllowedLogCatchupFrequencyInMinutes).MapKeys()),
+		))
+	}
+
+	if _, isPresent := api.AllowedWeeklySnapshotDays[tmInfo.WeeklySnapshotDay]; !isPresent {
+		*errors = append(*errors, field.Invalid(tmPath.Child("weeklySnapshotDay"), tmInfo.WeeklySnapshotDay,
+			fmt.Sprintf("Weekly Snapshot day must be specified. Valid values are: %s", reflect.ValueOf(api.AllowedWeeklySnapshotDays).MapKeys()),
+		))
+	}
+
+	if tmInfo.MonthlySnapshotDay < 1 || tmInfo.MonthlySnapshotDay > 28 {
+		*errors = append(*errors, field.Invalid(tmPath.Child("monthlySnapshotDay"), tmInfo.MonthlySnapshotDay, "Monthly snapshot day value must be between 1 and 28"))
+	}
+
+	if _, isPresent := api.AllowedQuarterlySnapshotMonths[tmInfo.QuarterlySnapshotMonth]; !isPresent {
+		*errors = append(*errors, field.Invalid(tmPath.Child("quarterlySnapshotMonth"), tmInfo.QuarterlySnapshotMonth,
+			fmt.Sprintf("Quarterly snapshot month must be specified. Valid values are: %s", reflect.ValueOf(api.AllowedQuarterlySnapshotMonths).MapKeys()),
+		))
+	}
+
+	if err := additionalArgumentsValidationCheck(spec.IsClone, instance.Type, instance.AdditionalArguments); err != nil {
+		*errors = append(*errors, field.Invalid(instancePath.Child("additionalArguments"), instance.AdditionalArguments, err.Error()))
+	}
+
+	databaselog.Info("Existing Database validateCreate logic!")
+}
+
+func initializeObjects(spec *DatabaseSpec) {
+	databaselog.Info(fmt.Sprintf("Entering initializeObjects logic..."))
+
+	// Initialize Database properties
+	if spec.Instance == nil {
```

Comment:
Should you be checking the type [new/clone] here to make sure you are not unnecessarily initializing clone input spec for db type and vice versa?

Reply:
We need to initialize all objects regardless of whether they are a db or clone type else web-hooks complains. It's odd why that is as the Clone and Instance type, as well as sub types are pointers.

Reply:
@mazin-s I feel we should still take a look why that causes errors, AFAIK when @akshmish introduced webhooks we did not face errors with nil pointer references. Not a P0 right now, but definitely something we should have an explanation for.

Reply:
Ok sounds good, I will revisit after.